src/Drawable.js

```diff
@@ -591,23 +606,27 @@ class Drawable {
      * @private
      */
     _getTransformedHullPoints () {
+        if (!this._transformedHullDirty) {
+            return this._transformedHullPoints;
+        }
+
         const projection = twgl.m4.ortho(-1, 1, -1, 1, -1, 1);
         const skinSize = this.skin.size;
         const halfXPixel = 1 / skinSize[0] / 2;
         const halfYPixel = 1 / skinSize[1] / 2;
         const tm = twgl.m4.multiply(this._uniforms.u_modelMatrix, projection);
-        const transformedHullPoints = [];
```

Comment:
Have you verified that callers never modify the contents of the array returned by this function?

Reply:
`_getTransformedHullPoints` is called by `getBounds` and `getBoundsForBubble`.

`getBounds` passes it to `Rectangle.initFromPointsAABB` which does not modify the array.

`getBoundsForBubble` calls `map` and `filter` on the array, both of which create new arrays instead of modifying in-place.